app/controllers/posts_controller.rb

```diff
@@ -8,4 +8,19 @@ def show
     @user = User.find(params[:user_id])
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+    @post = Post.new(user: @user, title: params[:post][:title], text: params[:post][:text])
+    if @post.save
+      redirect_to user_post_url({ user_id: @user.id, id: @post.id })
+      flash[:success] = 'Post created successfully'
+    else
+      render :new, flash: { error: 'Something went wrong with your post' }
+    end
+  end
```

Comment:
- [ ] Following rails design best practice. It will be nice to create a strong params method and place it under `private` and then call the method from there like so:

```
private 
    def post_params
      params.require(:post).permit(:title, :text)
    end
```
You can then use the method in the `create` action.